docs/docs/dev_docs/contracts/resources/common_patterns/main.md

```diff
@@ -0,0 +1,139 @@
+
+1. You pass the data as a parameter to your private method and later assert in public that the data is correct. E.g.:
+```rust
+struct Storage {
+   token: PublicState<Field, 1>,
+}
+
+contract Bridge {
+    
+    #[aztec(private)]
+    fn burn_token_private(
+        token: AztecAddress, // pass token here since this is a private method but can't access public storage
+        amount: Field,
+    ) -> Field {
+        ...
+    #include_code call_assert_token_is_same /yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr raw
+    }
+    #include_code assert_token_is_same /yarn-project/noir-contracts/src/contracts/token_bridge_contract/src/main.nr raw
+}
+```
+
+2. For public storage that changes infrequently, use the slow updates tree! More details TBD
+
+### When calling a public function from private, try to mark the public function as `internal`
+This ensures your flow works as intended and that no one can call the public function without going through the private function first!
+
+### Passing data from public to private
+You have public balance and want to move them into the private domain. You shouldn't pass your address that should receive the notes in private, because that leaks privacy (duh!). So what do you do?
+
+1. You have to create a note in public domain and can't encrypt it, because you can't leak the public key of the receiver. 
+2. So how do you control who can claim this note? Pass a hash of a secret instead of the address. And then in the private domain, pass the preimage (the secret) to later claim your funds
+
+So you have to create a custom note in the public domain that is not encrypted by some owner - we call such notes a "TransparentNote" since it is created in public, anyone can see the amount and the note is not encrypted by some owner.
+
+This pattern discussed in detail in [writing a token contract section in the shield() method](../../../tutorials/writing_token_contract.md#shield) and [redeem_shield() method](../../../tutorials/writing_token_contract.md#redeem_shield).
+
+### Discovering my notes
+When you send someone a note, the note hash gets encrypted and added to the merkle tree, but the receiver needs to know they receive a note so they can attempt to decrypt leaves in the tree and add it to their PXE. There are two ways you can discover your notes:
+
+1. When sending someone a note, use `emit_encrypted_log` (and encrypt the log with their public key). This way as the PXE processes all encrypted logs, it realizes there is a note for itself. [More info here](../../syntax/events.md)
+2.  Manually using `pxe.addNote()` - What if the contract doesn't emit such logs? Or you creating a note in the public domain and want to consume it in private domain (`emit_encrypted_log` shouldn't be called in the public domain because everything is public), like in the previous section where we created a TransparentNote in public. 
```

Comment:
> What if the contract doesn't emit such logs?

This is a bit weird as it's up to the developer to set this.

Instead I would describe here that a dev might be inclined here to not emit the logs to save gas.